add convenience bounding box type and bbox function

Crghilardi opened the pull request:
This fork add a new type `BoundingBox` and function `bbox` to `header.jl` as a way to quickly look at just the min/max values from the header. I used the `bbox` naming from the `sp`  R package.

I did not see this implemented in any other packages. It seems logical to have 

edit: forgot to add example result

`julia> bbox(header)`
`BoundingBox(568433.46, 569615.21, 5.04987947e6, 5.05083639e6, 1543.24, 1934.75)`

visr commented:
As you said, ideally we'd use a bounding box type from a more general geometry package. With this we have a new type, but no methods that work on it, therefore it is of little use. The best example I know is the [BoundingBox from RoamesGeometry.jl](https://github.com/FugroRoames/RoamesGeometry.jl/blob/af52a35c951bb8e71b363704ed624e117738c850/src/BoundingBox.jl). But in it's current state it is not something we can take a dependency on. Perhaps part of it can be integrated into GeometryTypes.jl or GeometryBasics.jl.

I do see your point about easily getting the bounding box from the LasHeader. How about a `boundingbox(header::LasHeader)` function that simply returns a NamedTuple? That also avoids having to remember the order of the parts, which tends to vary. Though since you have to choose anyway, perhaps it's good to go with the ordering as in the [GeoJSON spec](https://tools.ietf.org/html/rfc7946#section-5), which also seems to be the RoamesGeometry choice.

And I was thinking `boundingbox` instead of `bbox`, since it is more explicit, and then we can just use `bbox` as the variable name :)

What do you think?

Crghilardi commented:
I must have missed the `RoamesGeometry` package while I was clicking around, that looks like quite the package even with only one commit! 
Yes, I agree with the points you brought up. I thought it had value in terms of saving some typing and/or something a new user coming from another language may try to find.

I have not used NamedTuples since they were added to base Julia. 
If I am understanding the documentation correctly, I can remove the type and just have:

`boundingbox(h::LasHeader) = NamedTuple{(:x_min,:y_min,:z_min,:x_max,:y_max,:z_max)}((h.x_min,h.y_min,h.z_min,h.x_max,h.y_max,h.z_max))`

visr commented:
Yep, that's it. I would go for this syntax though: `(a=1, b=2)`

Crghilardi commented:
OK, I modified it to use NamedTuple and re-ordered outputs to match spec in latest commit

visr commented:
Ok thanks! Could you update the exports as well? The old bbox and BoundingBox are still there. And a little test to make sure it stays working would be great.

Crghilardi commented:
My mistake, those two items are now corrected. I added some tests in `runtests.jl`. I wasn't quite sure what to test on, so I copied a few examples from the archived [NamedTuple.jl package](https://github.com/JuliaData/NamedTuples.jl/blob/master/test/runtests.jl)

visr commented:
Thanks. I made some small notifications, didn't mean to close this. For future PR's, it's best to make them from a branch, not master. Are you ok with my changes in 94b707de0e0fc2d539eb9e417944e49e030378a2 (and my own fixup 2662dcc31b9340b3f76a699b5feb0e3e2cf6dc01)?

I changed the key names from `x_min` to `xmin`. It doesn't match with the `LasHeader` field name, but seems to match better with the rest of the world, like the linked RoamesGeometry package or for instance https://gdal.org/programs/gdalwarp.html. Plus it's shorter :)

And for the test I used `===` to test if they are identical. That also implies the same type and key order, as well as the values.

Crghilardi commented:
yes that all works. I figured using the same style as `::LasHeader` was a safe choice.
Good to know forks/branches for the future, hopefully I'll get better at this at some point. Thank you for your patience and assistance.
